pkg/debuginfo/store.go

```diff
@@ -228,9 +231,15 @@ func (s *Store) fetchObjectFile(ctx context.Context, buildID string) (string, er
 	// Check if it's already cached locally; if not download.
 	if _, err := os.Stat(mappingPath); os.IsNotExist(err) {
 		r, err := s.bucket.Get(ctx, path.Join(buildID, "debuginfo"))
+
 		if s.bucket.IsObjNotFoundErr(err) {
-			level.Debug(s.logger).Log("msg", "object not found", "object", buildID, "err", err)
-			return "", ErrDebugInfoNotFound
+			level.Debug(s.logger).Log("msg", "object not found in parca object storage", "object", buildID, "err", err)
+
+			r, err = s.debuginfodClientCache.GetDebugInfo(buildID)
```

Comment:
```suggestion
			r, err = s.debuginfodClientCache.GetDebugInfo(ctx, buildID)
```

As I pointed out above, context should be passed per request in here.